src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -172,82 +176,97 @@ private function getPendingMessages(): \Generator
                 'headers' => $headers,
             ];
         }
 
         $this->currentResponse = null;
 
-        yield from $this->getPendingMessages();
+        return true;
     }
 
     public function setup(): void
     {
-        $parameters = [
-            'Action' => 'CreateQueue',
+        $this->configuration['auto_setup'] = false;
+        if ($this->client->queueExists([
             'QueueName' => $this->configuration['queue_name'],
-        ];
+            'QueueOwnerAWSAccountId' => $this->configuration['account'],
+        ])->isSuccess()) {
+            return;
+        }
+
+        if (null !== $this->configuration['account']) {
+            throw new InvalidArgumentException(sprintf('The Amazon SQS queue "%s" does not exists (or you don\'t have permissions on it), and can\'t be created when an account is provided.', $this->configuration['queue_name']));
+        }
+
+        $parameters = ['QueueName' => $this->configuration['queue_name']];
 
-        if ($this->isFifoQueue($this->configuration['queue_name'])) {
+        if (self::isFifoQueue($this->configuration['queue_name'])) {
             $parameters['FifoQueue'] = true;
         }
 
-        $this->call($this->configuration['endpoint'], $parameters);
+        $this->client->createQueue($parameters)->resolve();
+        $exists = $this->client->queueExists(['QueueName' => $this->configuration['queue_name']]);
+        $exists->wait();
+        if (!$exists->isSuccess()) {
+            throw new TransportException(sprintf('Failed to crate the Amazon SQS queue "%s".', $this->configuration['queue_name']));
+        }
         $this->queueUrl = null;
-
-        $this->configuration['auto_setup'] = false;
     }
 
     public function delete(string $id): void
     {
-        $this->call($this->getQueueUrl(), [
-            'Action' => 'DeleteMessage',
+        $this->client->deleteMessage([
+            'QueueUrl' => $this->getQueueUrl(),
             'ReceiptHandle' => $id,
-        ]);
+        ])->resolve();
     }
 
     public function getMessageCount(): int
     {
-        $response = $this->request($this->getQueueUrl(), [
-            'Action' => 'GetQueueAttributes',
-            'AttributeNames' => ['ApproximateNumberOfMessages'],
+        $response = $this->client->getQueueAttributes([
+            'QueueUrl' => $this->getQueueUrl(),
+            'AttributeNames' => [QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES],
         ]);
-        $this->checkResponse($response);
-        $xml = new \SimpleXMLElement($response->getContent());
-        foreach ($xml->GetQueueAttributesResult->Attribute as $attribute) {
-            if ('ApproximateNumberOfMessages' !== (string) $attribute->Name) {
-                continue;
-            }
 
-            return (int) $attribute->Value;
-        }
+        $attributes = $response->getAttributes();
 
-        return 0;
+        return (int) ($attributes[QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES] ?? '0');
```

Comment:
Why use `"0"` when we cast it to int?

```suggestion
        return (int) ($attributes[QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES] ?? 0);
```

Reply:
thanks